src/Sylius/Bundle/ApiBundle/Resources/config/services.xml

```diff
@@ -99,6 +99,11 @@
             <tag name="sylius.uri_based_section_resolver" priority="40" />
         </service>
 
+        <service id="Sylius\Bundle\ApiBundle\EventListener\ChangePaymentMethodEventListener">
+            <argument type="service" id="Sylius\Component\Payment\Canceller\PaymentRequestCancellerInterface" />
+            <tag name="kernel.event_listener" event="kernel.controller_arguments" method="cancelPaymentRequestsWithDifferentPaymentMethod" />
```

Comment:
I'm not fully convinced to this listener as we listen for all controllers. There is a PaymentMethodChanger service and maybe we could use it to trigger cancelling requests? 🤔